app/controllers/tickets_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class TicketsController < ApplicationController
+  before_action :set_ticket, only: %i[show update]
+  before_action :set_contact, only: %i[update]
+  before_action :set_participating_seller, only: %i[create]
+
+  # GET /tickets
+  def index
+    @tickets = Ticket.all
+
+    json_response(@tickets)
+  end
+
+  # GET /tickets/1
+  def show
+    json_response(@ticket)
+  end
+
+  # POST /tickets
+  def create
+    savedCorrectly = true
+    createdTickets = []
+    numTix = params[:number_of_tickets]
+
+    if numTix.present? && numTix > 0
+      ActiveRecord::Base.transaction do
+        (1..numTix).each do
+          attributes = {}
+          attributes[:ticket_id] = Ticket.generate_ticket_id
+          attributes[:participating_seller] = @participating_seller
+
+          @ticket = Ticket.new(attributes)
+          savedCorrectly &&= @ticket.save!
+          createdTickets << @ticket
+        end
+      end
+    else
+      savedCorrectly = false
```

Comment:
I still don't think you need this. doing save! should raise an exception that will be caught by our exception_handler